### ISSUE TITLE
[Feature] New role that can comment, create and modify views, but not change data

### ISSUE THREAD

**Issue by pauljura** [nocodb/nocodb#2407]
**Need for this feature**

I'm looking at nocodb to give read-only access to data for non-developers. I'm happy for them to comment and create as many views as they like in order to gain as much insight into the data as they can. But I don't want them to modify the data itself. There is not yet a role designed specifically for this.

My workaround at the moment is to use a read-only database connection, but the UI still offers the ability to edit, it just displays error messages whenever someone tries to modify anything. It would be better if the ability to edit data was simply removed from the UI as it is in the "viewer" role, so they can freely click on anything without errors.

**Comment by o1lab** [comment]
Modifying view is higher permission than changing the data - hence the permissions as of now.

**Comment by dstala** [comment]
To gain insights into the data, they can still make use of sort & filters; edit error can be avoided for view role, we will work on it.

**Comment by ScriptJunky** [comment]
@dstala I don't believe that is an accurate statement. I have views I have shared that are in collaborative mode, and there is no difference with the locked mode. In both cases, sorting, filtering, and viewing images (1 cell, 2-3 images) is restricted.

**Comment by agbobeck** [comment]
I would like to add on a request for a user who can edit data and views, but not edit the base schema or have invite capabilities to the base. 

This essentially would be a additional permission level in between editor and creator, lets call it advanced editor for now

 #https://docs.nocodb.com/roles-and-permissions/roles-permissions-overview#table--view-operations

**Comment by dstala** [comment]
To be tracked under
[🔦 Feature: Custom roles](https://github.com/nocodb/nocodb/issues/2712)
